### DISCUSSION TITLE
Plugin on Ubuntu 24.04

### DISCUSSION THREAD

**Discussion by p4tt4nz** [apohl79/audiogridder#1370] [Help]
Tried to run the linux plugin on Ubuntu 24.04. Unfortunate its compiled on 20.04 so libcrypto and libssl is to old (1.1). 24.04 runs v3. I tried to just softlink but i got: version `OPENSSL_1_1_0'. Is it possible to get newer versions of the plugin that supports more up to date libs?

**Comment by tavasti** [comment]
Hit same problem in Devuan 5.0 (identical to debian 12)

**Comment by lumertzg** [comment]
Maybe you could install an older version of libssl alongside the newer one. This worked for me.

```sh
wget http://archive.ubuntu.com/ubuntu/pool/main/o/openssl/libssl1.1_1.1.0g-2ubuntu4_amd64.deb
sudo dpkg -i libssl1.1_1.1.0g-2ubuntu4_amd64.deb
```

**Comment by Olfried2** [comment]
I also have the problem on ubuntu 24.04 LTS but just installing the old lib makes me nervous ...
From -> https://gist.github.com/joulgs/c8a85bb462f48ffc2044dd878ecaa786
_"While it's always possible to install libssl1.1 if you needed it, note that OpenSSL 1.1 is [no longer being maintained by OpenSSL dev.](https://www.openssl.org/blog/blog/2023/03/28/1.1.1-EOL/) and can contain **unfixed security vulnerability**. If you're a dev, please consider migrating your program to OpenSSL 3. "_

The "link" solution seems to be best, but where does the plugin searches the lib? I have several old libs on my machine (not in the main path) to which I could link to. But where should I place the link?

**Comment by eu1444** [comment]
I was able to work around this using these steps:

1. Downloading libssl1.1 from the [previously mentioned gist](https://gist.github.com/joulgs/c8a85bb462f48ffc2044dd878ecaa786).
```bash
cd ~/Downloads
wget http://archive.ubuntu.com/ubuntu/pool/main/o/openssl/libssl1.1_1.1.0g-2ubuntu4_amd64.deb
```

2. Extracting the contents of the `.deb` file using the technique in this [Stack Overflow post](https://stackoverflow.com/a/53884160).
```bash
cd ~/Downloads
mkdir ~/libssl1.1
dpkg-deb -x libssl1.1_1.1.0g-2ubuntu4_amd64.deb ~/libssl1.1
```

3. Renaming `crashpad_handler` to `crashpad_handler.wrapped`.
```bash
sudo mv /usr/local/share/audiogridder/crashpad_handler /usr/local/share/audiogridder/crashpad_handler.wrapped
```

4. Creating a new wrapper script in place of the original `crashpad_handler`.
```bash
 sudo nano /usr/local/share/audiogridder/crashpad_handler
```

5.  Updating the script to set the `LD_LIBRARY_PATH` for loading the real handler, now named: `crashpad_handler.wrapped`. The `LD_LIBRARY_PATH` should point to wherever you stored the `libcrypto.so.1.1` and `libssl.so.1.1` that were extracted from the `.deb`.
```bash                                                                                          
#!/bin/bash
export LD_LIBRARY_PATH=~/libssl1.1/usr/lib/x86_64-linux-gnu:$LD_LIBRARY_PATH
/usr/local/share/audiogridder/crashpad_handler.wrapped "$@"
```

6. After that, all that was needed was to make the `crashpad_handler` wrapper script executable:
```bash
sudo chmod +x /usr/local/share/audiogridder/crashpad_handler
```

This method allows the `crashpad_handler` to use the old libssl without affecting any of the rest of the system.

**Reply by Olfried2** [reply]
This is the best way to do without security scare.
I got it working, but even my "windows server" is too slow (I tried to work with Native Instruments plugins) or it is not able to work with Harris Mixbus 8**. 
On my system the bash script does not work as supposed. I have to send the commands on the CLI as root by hand.
Maybe it's me...

Thank You for this solution.

**Comment by hermonculous** [comment]
I've been in the same situation and have, at lest in my case, resolved this by rebuilding on Ubuntu 24.04.  If this helps here are the steps...

run the sh install file to install the (not working) VSTs

**git clone --recurse-submodules https://github.com/apohl79/audiogridder.git**

**cd audiogridder**

**git clone https://github.com/apohl79/audiogridder-deps.git**   

Install any dependent libraries ( see build instructions for these)

Edit CMakeLists.txt line 9 to:
**option(AG_WITH_SERVER "Enable Server build." off)**

to stop build errors re. server build

Now execute:
**python3 build.py conf --disable-signing --type=Release**

..to build release rather than debug version.

then:
**python3 build.py build**

The built (VST3) files will be in audiogridder/build-linux-x86_64/debug (although actually release versions)

Copy the VST3 files to overwrite the files in ~/.vst3 - remove any other audiogridder plugins installed by the install sh file (in .vst folder)

copy file AudioGridderPluginTray to overwrite file in /usr/local/share/audiogridder

Hope I didn't miss anything but this is working fine for me now on Ubuntu 24.04...

**Comment by eufex** [comment]
I built it from Debian 12 and tested (plugin , not server ) on AV Linux MX and Ubuntu 24 also. It’s shared from my Gdrive for all to use and has been since October 2024. Forgive me for not posting the link here, but off the top of my head, I have no idea what it is. The archive also contains a text file instructing where to manually place the files as when you build, there’s no installer.
